Lib/gftools/builder/_ninja.py

```diff
@@ -0,0 +1,360 @@
+            axis_tags = ",".join(axis_tags)
+            target = os.path.join(
+                self.config["vfDir"],
+                Path(designspace_path).stem + "[%s].ttf" % axis_tags,
+            )
+            self.w.build(
+                target,
+                "variable",
+                designspace_path,
+                variables={
+                    "fontmake_args": self.fontmake_args({"output_path": target})
+                },
+            )
+            targets.append(target)
+        self.w.newline()
+        stampfile = self.gen_stat(axis_tags, targets)
+        # We post process each variable font after generating the STAT tables
+        # because these tables are needed in order to fix the name tables.
+        self.w.comment("Variable font post-processing")
+        for t in targets:
+            self.post_process_variable(t, implicit=stampfile)
+
+    def gen_stat(self, axis_tags, targets):
+        self.w.comment("Generate STAT tables")
+        if "axisOrder" not in self.config:
+            self.config["axisOrder"] = axis_tags.split(",")
+            # Janky "is-italic" test. To strengthen this up we should look inside
+            # the source files and check their stylenames.
+            if any("italic" in x[0].lower() for x in self.designspaces):
+                self.config["axisOrder"].append("ital")
+        other_args = ""
+        stampfile = targets[0] + ".statstamp"
+        if "stat" in self.config:
+            statfile = targets[0] + ".stat.yaml"
+            os.makedirs(os.path.dirname(statfile), exist_ok=True)
+            open(statfile, "w").write(yaml.dump(self.config["stat"]))
+            other_args = f"--src {statfile}"
+        if "stylespaceFile" in self.config or "statFormat4" in self.config:
+            raise ValueError(
+                "Stylespace files / statFormat4 not supported in Ninja mode"
+            )
+            # Because gftools-gen-stat doesn't seem to support it?
+        self.temporaries.append(stampfile)
+        self.w.build(
+            stampfile,
+            "genstat",
+            targets,
+            variables={
+                "other_args": other_args,
+                "stampfile": stampfile,
+            },
+        )
+        self.w.newline()
+        return stampfile
+
+    def post_process(self, file, implicit=None):
+        variables = {}
+        if self.config["includeSourceFixes"]:
+            variables = {"fixargs": "--include-source-fixes"}
+        self.temporaries.append(file + ".fixstamp")
+        self.w.build(
+            file + ".fixstamp", "fix", file, implicit=implicit, variables=variables
+        )
+
+    def _instance_ufo_filenames(self, path, designspace):
+        instance_filenames = []
+        for instance in designspace.instances:
+            fn = instance.filename
+            if "/" in fn:
+                ufo = Path(fn)
+            else:
+                ufo = Path(path).parent / fn
```

Comment:
the logic here is incorrect, instance.filename is always meant to be interpreted as relative to the designspace's directory (the `else:` branch above); checking `if "/" in fn` doesn't make any sense, instance.filename can be a simple filename without any path separator (meaning in the same directory as the DS), or it can be a subdirectory relative to the DS directory (thus containing a "/" which should not be ignored).